extensions/ql-vscode/src/databases-ui.ts

```diff
@@ -101,19 +100,19 @@ class DatabaseTreeDataProvider extends DisposableObject
     return this._onDidChangeTreeData.event;
   }
 
-  private handleDidChangeDatabaseItem = (
-    databaseItem: DatabaseItem | undefined
-  ): void => {
-    this._onDidChangeTreeData.fire(databaseItem);
+  private handleDidChangeDatabaseItem = (event: DatabaseChangedEvent): void => {
+    // Note that events from the databse manager are instances of DatabaseChangedEvent
+    // and events fired by the UI are instances of DatabaseItem
```

Comment:
Will we ever receive a `DatabaseItem` here, and so need to handle `event.item || event as DatabaseItem`?

Reply:
`item` is an optional field on the event. The `onDidChangeTreeData` method is from the super class. When that method is called, this triggers the tree view to update. If an item is passed, then only that item (and children, but db view items have no children) is updated. If no item is passed, then the entire tree is updated. 

When we add or remove an item, the entire tree needs to be updated and when an item is renamed, we only need to update that single item.

Hopefully, that clears things up a little. I'll add a comment on that matter.

Reply:
Oh got it - `event.item` will just be `undefined` if it's not actually a `DatabaseItem`. Was thinking too much in Java!